README.md

```diff
@@ -181,6 +181,112 @@ If you want to compile with ROCm support, install
 If you want to disable ROCm support, export the environment variable `USE_ROCM=0`.
 Other potentially useful environment variables may be found in `setup.py`.
 
+---
+## Comprehensive Tutorial: Compiling PyTorch from Source and Installing CUDA Deep Learning Libraries on Ubuntu WSL2
+This tutorial provides a step-by-step guide to compiling PyTorch from source and installing CUDA deep learning libraries on Ubuntu 20.04. It includes instructions on updating and upgrading Ubuntu, installing Anaconda, setting up a Python environment, and installing NVIDIA CUDA and cuDNN.
+### Prerequisites
+- Ubuntu OS
+- NVIDIA GPU
+- Basic knowledge of using the terminal
+### Step 1: Update and Upgrade Ubuntu
+Update and upgrade your Ubuntu system to ensure you have the latest packages and security updates.
+```bash
+sudo apt update
+sudo apt dist-upgrade
+```
+### Step 2: Set Up Downloads Directory (Optional)
+Create a symbolic link to your Downloads directory if you have one on your Windows system.
+```bash
+ln -s /mnt/c/Users/<user>/Downloads Downloads
+```
+Replace `<user>` with your Windows username.
+### Step 3: Install Anaconda
```

Comment:
Why Anaconda installation is necessary? What is wrong with system pytorch available on Ubuntu?

Reply:
Each application  I am running is asking for a specific pytorch version. for example today I installed a triton Nvidia server package  and it downgraded pytorch to version 2.2 and cuda 11. but installing  into a conda environment  it is easier to manage other dependencies, also I was unable to download a version to work with cuda 12.4 from any repo. I think the supplied ones are max 12.1 cuda.

Reply:
can you explain what you mean here "Why Anaconda installation is necessary?" because in the original readme it says "We highly recommend installing an [Anaconda](https://www.anaconda.com/download) environment. You will get a high-quality BLAS library (MKL) and you get controlled dependency versions regardless of your Linux distro." and also "conda install intel::mkl-static intel::mkl-include" "conda install -c pytorch magma-cuda124 " these steps where already done in that original readme. yet i will install into the tutorial. I have created a new ubuntu fresh install to test the full procedure. yet i am unsure on your advice here is to install pytorch for cuda 12.4 compiled here as the system python version which would need to be updated to 3.12 as a new part of the tutorial. or if you are recommending to install in the conda base environment. instead of the pytorch-dev one.